File: pinyougou-sellergoods-interface/src/main/java/cn/pinyougou/sellergoods/service/BrandService.java
package cn.pinyougou.sellergoods.service;

import cn.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

/**
 * 品牌接口
 */
public interface BrandService {
    /**
     * 查询所有
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 返回分页列表
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);

    /**
     * 添加数据
     * @param brand
     */
    public void add(TbBrand brand);

    /**
     * 根据id查询实体
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 修改数据
     * @param brand
     */
    public void update(TbBrand brand);

    /**
     * 删除数据
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页查询
     * @param brand
     * @param pageNum 当前页码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//服务抽取 自定义服务
app.service("brandService", function ($http) {
    //读取数据列表绑定到订单中
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };
    //分页服务
    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    };
    //根据id查询
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };
    //添加数据
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity)
    };
    //修改数据
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity)
    };
    //删除数据
    this.dele = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    };
    //条件查询
    this.search = function (page, size, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, searchEntity);
    };


});